consumer-java/aws-kinesis-video-producer-sdk-canary-consumer/src/main/java/com/amazon/kinesis/video/canary/consumer/Terminate.java

```diff
@@ -0,0 +1,30 @@
+package com.amazon.kinesis.video.canary.consumer;
+
+import com.amazonaws.kinesisvideo.parser.examples.ContinuousGetMediaWorker;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+
+@Slf4j
+public class Terminate extends ProducerSdkCanaryConsumer implements Runnable {
+   public void run() {
+   	Integer canaryRunTime = Integer.parseInt(System.getenv("CANARY_DURATION_IN_SECONDS"));
+      System.out.println("Canary Run time: " + canaryRunTime);
+      try 
+		{
+   		Thread.sleep(canaryRunTime * 1000);
```

Comment:
Why are we waiting?

Reply:
We want to stop the application after run duration